### PR TITLE
Adopt PG 15 behavior for CREATE privileges

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-8107

Summary of changes:

- Update `CREATE` privileges docs with new cluster setting `sql.auth.public_schema_create_privilege.enabled` for compatibility with the recent Postgres 15 release.

### CODE REVIEW

**Comment by rmloveland** [comment]
@rafiss please let me know if more is needed here, this is possibly a "minimum viable change" to the docs but maybe there's not quite enough to be viable!

File: src/current/_includes/v23.2/sql/privileges.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Privilege | Levels | Description
...
+`CREATE` | Database, Schema, Table, Sequence | Grants the ability to create objects at the database, schema, table, or sequence level. When applied at the database level, grants the ability to configure [multi-region zone configs]({% link {{ page.version.version }}/zone-config-extensions.md %}). <br /><br /> In CockroachDB v23.2 and later, the [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) `sql.auth.public_schema_create_privilege.enabled`. This setting controls whether users receive `CREATE` privileges on the public schema or not. The setting applies at the time that the [public schema is created]({% link {{ page.version.version }}/create-schema.md %}), which happens whenever [a database is created]({% link {{ page.version.version }}/create-database.md %}). The setting is `true` by default, but can be set to `false` for increased compatibility with [Postgres version 15](https://www.postgresql.org/about/news/postgresql-15-released-2526/) as described in [this commit](https://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=b073c3ccd06e4cb845e121387a43faa8c68a7b62).
```
Comment (rafiss) [inline src/current/_includes/v23.2/sql/privileges.md:8]: i don't think anywhere else needs changes. but one quick edit here:
> In CockroachDB v23.2 and later, the [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) `sql.auth.public_schema_create_privilege.enabled`. This setting controls whether users receive `CREATE` privileges on the public schema or not.

should be 
> In CockroachDB v23.2 and later, the [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) `sql.auth.public_schema_create_privilege.enabled` controls whether users receive `CREATE` privileges on the public schema or not.

**Review by rafiss — APPROVED** [review]

File: src/current/_includes/v23.2/sql/privileges.md (hunk 1 above)
Comment (rmloveland) [reply src/current/_includes/v23.2/sql/privileges.md:8]: thanks @rafiss ! guess that typo snuck in via some edits :-) 

this is updated now to your suggested text

File: src/current/_includes/v23.2/sql/privileges.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ Privilege | Levels | Description
...
+`CREATE` | Database, Schema, Table, Sequence | Grants the ability to create objects at the database, schema, table, or sequence level. When applied at the database level, grants the ability to configure [multi-region zone configs]({% link {{ page.version.version }}/zone-config-extensions.md %}). <br /><br /> In CockroachDB v23.2 and later, the [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) `sql.auth.public_schema_create_privilege.enabled` controls whether users receive `CREATE` privileges on the public schema or not. The setting applies at the time that the [public schema is created]({% link {{ page.version.version }}/create-schema.md %}), which happens whenever [a database is created]({% link {{ page.version.version }}/create-database.md %}). The setting is `true` by default, but can be set to `false` for increased compatibility with [Postgres version 15](https://www.postgresql.org/about/news/postgresql-15-released-2526/) as described in [this commit](https://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=b073c3ccd06e4cb845e121387a43faa8c68a7b62).
```
Comment (taroface) [inline src/current/_includes/v23.2/sql/privileges.md:8]: nit: Postgres -> PostgreSQL

```suggestion
`CREATE` | Database, Schema, Table, Sequence | Grants the ability to create objects at the database, schema, table, or sequence level. When applied at the database level, grants the ability to configure [multi-region zone configs]({% link {{ page.version.version }}/zone-config-extensions.md %}). <br /><br /> In CockroachDB v23.2 and later, the [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) `sql.auth.public_schema_create_privilege.enabled` controls whether users receive `CREATE` privileges on the public schema or not. The setting applies at the time that the [public schema is created]({% link {{ page.version.version }}/create-schema.md %}), which happens whenever [a database is created]({% link {{ page.version.version }}/create-database.md %}). The setting is `true` by default, but can be set to `false` for increased compatibility with [PostgreSQL version 15](https://www.postgresql.org/about/news/postgresql-15-released-2526/) as described in [this commit](https://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=b073c3ccd06e4cb845e121387a43faa8c68a7b62).
```

**Review by taroface — APPROVED** [review]
LGTM w/ a nit!

File: src/current/_includes/v23.2/sql/privileges.md (hunk 2 above)
Comment (rmloveland) [reply src/current/_includes/v23.2/sql/privileges.md]: Fixed in latest push - thanks!

**Pull request merged**